### ISSUE TITLE
change event.tags to many-to-many relationship

### ISSUE THREAD

**Issue by hburgund** [roundware/roundware-server#227]
Currently the `event` model has a `tags` field which is simply a comma-delimited string of tag ids.  This is essentially un-searchable and therefore not terribly useful.  It is also not at all good practice.

Presumably the best fix is to remove the `event.tags` field and replace it with a many-to-many relationship table with the tags table.  But I'm all ears for other suggestions.

This is nothing that needs to happen anytime soon; I'm just creating this so we don't forget.

**Comment by grayb** [comment]
Agreed.  The first time I looked at this, I had the thought that it was intentionally this way because you might have a need to log an event with a tag that might not exist in the future.  If perhaps the tag was deleted, then you would have no way in the future to reference the tag that was part of this timed, in the past, event.  Given that assumption though, you'd want to store the text of the tag, and not the ID.

It is something to consider if tags will be delete-able in the future.

**Comment by sinabahram** [comment]
If that need exists, we could always mark as deleted, modifying queries appropriately, or more performantly, simply move to a deleted tags table.

**Comment by hburgund** [comment]
I can see no reason why tags should ever be deletable without consequences, so I'm not worried about building anything specific to allow for this.  Tags can easily be deactivated without being deleted which should handle any reasonable situation as best I can tell.  If I am proven wrong, we now have some good possible solutions though!
